src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+    return false;
+  }
+  LM_M(("KZ: Service-Path OK"));
+
+
+  //
+  // If ONCHANGE and one of the attribute names in the scope vector
+  // of the subscription has the same name as the incoming attribute. there is a match.
+  //
+  LM_M(("KZ: Matching Attributes (incoming: %s)", attributeName.c_str()));
+  if (!attributeMatch(attributeName))
+  {
+    LM_M(("KZ: Subscription::match: attributes not matching (incoming: '%s')", attributeName.c_str()));
+    return false;
+  }
+  LM_M(("KZ: Attributes OK"));
+
+  for (unsigned int ix = 0; ix < entityIdInfos.size(); ++ix)
+  {
+    if (entityIdInfos[ix]->match(id, type))
+    {
+      LM_M(("KZ: Subscription::match: WE HAVE A MATCH !!!"));
+      return true;
+    }
+  }
+
+  LM_M(("KZ: Subscription::match: entityId not matching"));
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* match - 
+*/
+bool Subscription::servicePathMatch
+(
+  const std::string&  _servicePath
+)
+{
+  const char* spath = servicePath.c_str();
+
+  //
+  // No wildcard - exact match
+  //
+  if (spath[strlen(spath) - 1] != '#')
+  {
+    LM_M(("KZ: No wildcard - exact match necessary"));
+    return (_servicePath == servicePath);
+  }
+
+  // Special case. "/#" matches EVERYTHING
+  if (servicePath == "/#")
+  {
+    LM_M(("KZ: Wildcard MATCH 1"));
+    return true;
+  }
+
+
+  //
+  // Wildcard - remove '#' and compare to the length of _servicePath.
+  //            If equal upto the length of _servicePath, then it is a match
+  //
+  // Actually, there is more than one way to match here:
+  // If we have a servicePath of the subscription as 
+  // "/a/b/#", then the following service-paths must match:
+  //   1. /a/b
+  //   2. /a/b/ AND /a/b/.+  (any path below /a/b/)
+  //
+  // What should NOT match is "/a/b2".
+  //
+  unsigned int len = strlen(spath) - 2;
+
+  // 1. /a/b - (removing 2 chars from the cache-subscription removes "/#"
+
+  if ((spath[len] == '/') && (strlen(_servicePath.c_str()) == len) && (strncmp(spath, _servicePath.c_str(), len) == 0))
+  {
+    LM_M(("KZ: Wildcard MATCH 2"));
+    return true;
+  }
+
+  // 2. /a/b/.+
+  len = strlen(spath) - 1;
+  if (strncmp(spath, _servicePath.c_str(), len) == 0)
+  {
+    LM_M(("KZ: Wildcard MATCH 3"));
+    return true;
+  }
+  
+  LM_M(("KZ: Wildcard NO MATCH"));
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* hasAttribute - 
+*/
+bool Subscription::hasAttribute(const std::string& attributeName)
+{
+  unsigned int ix;
+
+  for (ix = 0; ix < attributes.size(); ++ix)
+  {
+    if (attributes[ix] == attributeName)
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* update - 
+*/
+void Subscription::update(UpdateContextSubscriptionRequest* ucsrP)
```

Comment:
Yeah, I was thinking the same. Especially as this implements "patch" and not "put" and what we want is a 'put-behaviour', right?